lms/static/scripts/frontend_apps/utils/grade-validation.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * Result of parsing a grade value entered by the user.
+ *
+ * This is either a grade value or a validation error message.
+ */
+export type ValidateGradeResult =
+  | { ok: false; error: string }
+  | { ok: true; grade: number };
```

Comment:
Perhaps a `valid` prop instead of an `ok` prop gives a bit more context?

```suggestion
export type ValidateGradeResult =
  | { valid: false; error: string }
  | { valid: true; grade: number };
```

Reply:
I agree. Will do.